activerecord/test/cases/adapters/sqlite3/sqlite3_adapter_test.rb

```diff
@@ -136,6 +136,26 @@ def test_encoding
         assert_equal "UTF-8", @conn.encoding
       end
 
+      def test_default_pragmas_for_file_connection
+        with_file_connection do |conn|
+          assert_equal [{ "foreign_keys" => 1 }], conn.execute("PRAGMA foreign_keys")
+          assert_equal [{ "journal_mode" => "wal" }], conn.execute("PRAGMA journal_mode")
+          assert_equal [{ "synchronous" => 1 }], conn.execute("PRAGMA synchronous")
+          assert_equal [{ "journal_size_limit" => 67108864 }], conn.execute("PRAGMA journal_size_limit")
+          assert_equal [{ "mmap_size" => 134217728 }], conn.execute("PRAGMA mmap_size")
+          assert_equal [{ "cache_size" => 2000 }], conn.execute("PRAGMA cache_size")
+        end
+      end
+
+      def test_default_pragmas_for_memory_connection
+        assert_equal [{ "foreign_keys" => 1 }], @conn.execute("PRAGMA foreign_keys")
+        assert_equal [{ "journal_mode" => "memory" }], @conn.execute("PRAGMA journal_mode")
+        assert_equal [{ "synchronous" => 2 }], @conn.execute("PRAGMA synchronous")
+        assert_equal [{ "journal_size_limit" => 67108864 }], @conn.execute("PRAGMA journal_size_limit")
+        assert_equal [], @conn.execute("PRAGMA mmap_size")
+        assert_equal [{ "cache_size" => 2000 }], @conn.execute("PRAGMA cache_size")
+      end
```

Comment:
Would it make sense for this to be a single test where the assertions are conditional based on `in_memory_db?` Buildkite runs this whole file against both regular and memory configurations.

Reply:
Good point. Updated.